WooCommerce/Classes/POS/Presentation/Card Present Payments/PointOfSaleCardPresentPaymentEventPresentationStyle.swift

```diff
@@ -0,0 +1,114 @@
+                    error: error,
+                    cancelAction: endSearch)))
+
+        case .connectingFailedUpdatePostalCode(let retrySearch, let endSearch):
+            return .alert(.connectingFailedUpdatePostalCode(
+                viewModel: PointOfSaleCardPresentPaymentConnectingFailedUpdatePostalCodeAlertViewModel(
+                    retryButtonAction: retrySearch,
+                    cancelButtonAction: endSearch)))
+
+        case .connectingFailedChargeReader(let retrySearch, let endSearch):
+            return .alert(.connectingFailedChargeReader(
+                viewModel: PointOfSaleCardPresentPaymentConnectingFailedChargeReaderAlertViewModel(
+                    retryButtonAction: retrySearch,
+                    cancelButtonAction: endSearch)))
+
+        case .connectingFailedUpdateAddress(let wcSettingsAdminURL, let retrySearch, let endSearch):
+            return .alert(.connectingFailedUpdateAddress(
+                viewModel: PointOfSaleCardPresentPaymentConnectingFailedUpdateAddressAlertViewModel(
+                    settingsAdminUrl: wcSettingsAdminURL,
+                    retrySearchAction: retrySearch,
+                    cancelSearchAction: endSearch)))
+
+        case .foundReader(let name, let connect, let continueSearch, let endSearch):
+            return .alert(.foundReader(
+                viewModel: PointOfSaleCardPresentPaymentFoundReaderAlertViewModel(
+                    readerName: name,
+                    connectAction: connect,
+                    continueSearchAction: continueSearch,
+                    endSearchAction: endSearch)))
+
+        case .updateProgress(let requiredUpdate, let progress, let cancelUpdate):
+            return .alert(.updatingReader(viewModel: CardPresentPaymentUpdatingReaderAlertViewModel()))
+
+        case .updateFailed(let tryAgain, let cancelUpdate):
+            return .alert(.updateFailed(viewModel: CardPresentPaymentReaderUpdateFailedAlertViewModel()))
+
+        case .updateFailedNonRetryable(let cancelUpdate):
+            return .alert(.updateFailed(viewModel: CardPresentPaymentReaderUpdateFailedAlertViewModel()))
+
+        case .updateFailedLowBattery(let batteryLevel, let cancelUpdate):
+            return .alert(.updateFailed(viewModel: CardPresentPaymentReaderUpdateFailedAlertViewModel()))
+
+        /// Payment messages
+        case .preparingForPayment(cancelPayment: let cancelPayment):
+            return .message(.preparingForPayment)
+
+        case .tapSwipeOrInsertCard(inputMethods: let inputMethods, cancelPayment: let cancelPayment):
+            return .message(.tapSwipeOrInsertCard)
+
+        case .success(done: let done):
+            return .message(.success)
+
+        case .error(error: let error, tryAgain: let tryAgain, cancelPayment: let cancelPayment):
+            return .message(.error)
+
+        case .errorNonRetryable(error: let error, cancelPayment: let cancelPayment):
```

Comment:
nit: similar to the `error` case above, maybe this case name could provide more context.